Libraries/Image/RCTImageLoader.m

```diff
@@ -779,6 +779,25 @@ - (RCTImageLoaderCancellationBlock)getImageSizeForURLRequest:(NSURLRequest *)ima
                                 completionBlock:completion];
 }
 
+- (NSDictionary *)getImageCacheStatus:(NSArray *)requests
+{
+  NSMutableDictionary *results = [NSMutableDictionary dictionary];
+  for (id request in requests) {
+    NSURLRequest *urlRequest = [RCTConvert NSURLRequest:request];
+    if (urlRequest) {
+      NSCachedURLResponse *cachedResponse = [NSURLCache.sharedURLCache cachedResponseForRequest:urlRequest];
+      if (cachedResponse) {
+        if (cachedResponse.storagePolicy == NSURLCacheStorageAllowedInMemoryOnly) {
+          [results setObject:@"memory" forKey:urlRequest.URL.absoluteString];
+        } else {
+          [results setObject:@"disk" forKey:urlRequest.URL.absoluteString];
```

Comment:
>[results setObject:@"disk" forKey:urlRequest.URL.absoluteString]; [](start = 10, length = 65)

Do we want this branch to run when `storagePolicy` is `NSURLCacheStorageNotAllowed`?

Reply:
My understanding is that responses obtained from the cache will never have such ```storagePolicy``` because they would not be inserted in the cache in the first place. I recognize that it would be safer to explicitly check for ```NSURLCacheStorageAllowed``` in the second if, though.